pkg/kubevirt/utils.go

```diff
@@ -138,36 +143,90 @@ func buildVirtualMachineInstanceTemplate(ctx *context.MachineContext) *kubevirtv
 
 	template.Spec = *ctx.KubevirtMachine.Spec.VirtualMachineTemplate.Spec.Template.Spec.DeepCopy()
 
-	cloudInitVolumeName := "cloudinitvolume"
-	cloudInitVolume := kubevirtv1.Volume{
-		Name: cloudInitVolumeName,
-		VolumeSource: kubevirtv1.VolumeSource{
-			CloudInitConfigDrive: &kubevirtv1.CloudInitConfigDriveSource{
-				UserDataSecretRef: &corev1.LocalObjectReference{
-					Name: *ctx.Machine.Spec.Bootstrap.DataSecretName + "-userdata",
+	cloudInitType, index := detectCloudInitVolumeType(template)
+	cloudInitVolume := cloudinitVolume(template, ctx, cloudInitType)
+	cloudInitDisk := cloudinitDisk(template, cloudInitType)
+	switch cloudInitType {
+	case "CloudInitNoCloud":
+		err := mergo.Merge(&template.Spec.Volumes[index], cloudInitVolume)
+		if err != nil {
+			panic(err)
+		}
+		break
```

Comment:
Please remove. Not needed on golang. It will aldo solve the linter warning.

Reply:
Updated in e58acaa